wicker/core/storage.py

```diff
@@ -463,17 +487,23 @@ class S3PathFactory(WickerPathFactory):
                     - part-1-attempt-2345.parquet
     """
 
-    def __init__(self, s3_root_path: Optional[str] = None) -> None:
+    def __init__(self, s3_root_path: Optional[str] = None, prefix_replace_path: str = "") -> None:
         """Init S3PathFactory.
 
         Args:
             s3_root_path (Optional[str], optional): path to s3 root path. Defaults to None.
+            prefix_replace_pathm (str, optional): path to replace for given prefix. In this class this is generally
+                replacing s3 when a bucket is mounted to instance. Defaults to "".
         """
         s3_config = get_config().aws_s3_config
         store_concatenated_bytes_files_in_dataset = s3_config.store_concatenated_bytes_files_in_dataset
         s3_root_path = s3_root_path if s3_root_path is not None else s3_config.s3_datasets_path
         # ignore type as we already handled none case above
-        super().__init__(s3_root_path, store_concatenated_bytes_files_in_dataset)  # type: ignore
+        super().__init__(
+            prefix_replace_path=prefix_replace_path,
+            root_path=s3_root_path,
```

Comment:
Super nit would you mind passing `root_path` as the first argument since in the base class it does not declare a default value (i.e. it is an arg). Just a good coding practice here in case the base class `__init__` function args and kwargs ever change further.

Reply:
Done, wasn't sure when they're all passed as kwargs even though they dont all have defaults, thanks for pointing that out.